test/middleware/CompanyAuthMiddleware.test.ts

```diff
@@ -0,0 +1,155 @@
+        sessionCookieName: '__SID',
+        sessionCookieDomain: 'rebel1.aws.chdev.org',
+        sessionCookieSecureFlag: 'true',
+        sessionTimeToLiveInSeconds: 3600
+    };
+
+    const featureFlag: string = '1';
+
+    it('should call next with an error when session is undefined', async () => {
+
+        const encryptionService = createEncryptionService('resolves');
+
+        const sessionStore = Substitute.for<SessionStore>();
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = { session: undefined } as Request;
+
+        try{
+            await companyAuthMiddleware.handler(request, response, nextFunction);
+        } catch (err) {
+            expect(err.message).to.equal(SESSION_NOT_FOUND_ERROR.message);
+        }
+    });
+
+
+    it('should call next if the user is authorized for company number', async () => {
+
+        const appData = { appeal };
+
+        const encryptionService = createEncryptionService('resolves');
+
+        const sessionStore = Substitute.for<SessionStore>();
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = getRequestSubstitute(appData, appeal.penaltyIdentifier.companyNumber);
+
+        await companyAuthMiddleware.handler(request, response, nextFunction);
+        nextFunction.received(1);
+        response.didNotReceive().redirect(Arg.any());
+
+    });
+
+    it('should redirect if the user is not authorized for company number', async () => {
+
+        const appData = { appeal };
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = getRequestSubstitute(appData, '');
+
+        const encryptionService = createEncryptionService('resolves', 'a_sequence');
+
+        const sessionStore = Substitute.for<SessionStore>();
+
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        await companyAuthMiddleware.handler(request, response, nextFunction);
+        nextFunction.didNotReceive();
+        response.received(1).redirect(Arg.any());
+
+    });
+});
+
+const createEncryptionService = (method: 'resolves' | 'rejects', encoded?: string | any)
```

Comment:
Not a big fan of functions like that. Feels strange to have to say resolve / reject as via string arg. Here is how we used to do it before:

```
const emailService = createSubstituteOf<EmailService>(service => {
    service.send(Arg.any()).returns(Promise.reject(Error('Unexpected error')));
});
```

In above example it is clear to me what substitute of email service is setup to do.